core/src/test/java/org/apache/iceberg/TestBase.java

```diff
@@ -0,0 +1,772 @@
+          .withPath("/path/to/data-b-deletes.parquet")
+          .withFileSizeInBytes(10)
+          .withPartitionPath("data_bucket=1") // easy way to set partition data for now
+          .withRecordCount(1)
+          .build();
+  static final DataFile FILE_C =
+      DataFiles.builder(SPEC)
+          .withPath("/path/to/data-c.parquet")
+          .withFileSizeInBytes(10)
+          .withPartitionPath("data_bucket=2") // easy way to set partition data for now
+          .withRecordCount(1)
+          .withSplitOffsets(ImmutableList.of(2L, 8L))
+          .build();
+  static final DeleteFile FILE_C2_DELETES =
+      FileMetadata.deleteFileBuilder(SPEC)
+          .ofEqualityDeletes(1)
+          .withPath("/path/to/data-c-deletes.parquet")
+          .withFileSizeInBytes(10)
+          .withPartitionPath("data_bucket=2") // easy way to set partition data for now
+          .withRecordCount(1)
+          .build();
+  static final DataFile FILE_D =
+      DataFiles.builder(SPEC)
+          .withPath("/path/to/data-d.parquet")
+          .withFileSizeInBytes(10)
+          .withPartitionPath("data_bucket=3") // easy way to set partition data for now
+          .withRecordCount(1)
+          .withSplitOffsets(ImmutableList.of(0L, 3L, 6L))
+          .build();
+  static final DeleteFile FILE_D2_DELETES =
+      FileMetadata.deleteFileBuilder(SPEC)
+          .ofEqualityDeletes(1)
+          .withPath("/path/to/data-d-deletes.parquet")
+          .withFileSizeInBytes(10)
+          .withPartitionPath("data_bucket=3") // easy way to set partition data for now
+          .withRecordCount(1)
+          .build();
+  static final DataFile FILE_WITH_STATS =
+      DataFiles.builder(SPEC)
+          .withPath("/path/to/data-with-stats.parquet")
+          .withMetrics(
+              new Metrics(
+                  10L,
+                  ImmutableMap.of(3, 100L, 4, 200L), // column sizes
+                  ImmutableMap.of(3, 90L, 4, 180L), // value counts
+                  ImmutableMap.of(3, 10L, 4, 20L), // null value counts
+                  ImmutableMap.of(3, 0L, 4, 0L), // nan value counts
+                  ImmutableMap.of(
+                      3,
+                      Conversions.toByteBuffer(Types.IntegerType.get(), 1),
+                      4,
+                      Conversions.toByteBuffer(Types.IntegerType.get(), 2)), // lower bounds
+                  ImmutableMap.of(
+                      3,
+                      Conversions.toByteBuffer(Types.IntegerType.get(), 5),
+                      4,
+                      Conversions.toByteBuffer(Types.IntegerType.get(), 10)) // upperbounds
+                  ))
+          .withFileSizeInBytes(350)
+          .build();
+
+  static final FileIO FILE_IO = new TestTables.LocalFileIO();
+
+  @TempDir protected Path temp;
+
+  @TempDir protected File tableDir = null;
+  protected File metadataDir = null;
+  public TestTables.TestTable table = null;
+
+  protected final int formatVersion;
```

Comment:
with https://github.com/apache/iceberg/pull/9161 I think this should be marked as `@Parameter` rather than having a constructor that sets this

Reply:
The `formatVersion` is used for create different `TableAssertions` in the constructor, so I think it need to keep using constructor to init the `formatVersion`. And in this way `ParameterizedTestExtension` can work well too.

Reply:
OK, I will convert some example test classes use the TestBase. Should I commit to this pr?

Reply:
> OK, I will convert some example test classes use the TestBase. Should I commit to this pr?

BTW, the classes using the base test depend on #9161, can I directly merge his code or wait for the pr?

Reply:
let's wait until we have `ParameterizedTestExtension` on main.